### ISSUE TITLE
No idempotency when using secret parameters

### ISSUE THREAD

**Issue by markgoddard** [pulp/squeezer#88]
# Steps to reproduce

* Run a task that creates a resource with a secret field, e.g. a remote with a client_key.
* Run the task again.

# Expected

The second task status is ok

# Actual

The second task status is changed

# Analysis

This seems to happen because the client_key is not returned by the API when the object is queried.

**Comment by mdellweg** [comment]
That is correct. And there is no way the client tooling can know whether something changed. So given any secret, we always POST the entity and need to assume we changed sth.
